### DISCUSSION TITLE
Having issues with the "peripheral.getType" method

### DISCUSSION THREAD

**Discussion by MintTee** [cc-tweaked/CC-Tweaked#2022] [Q&A] [answered]
I don't understand how to actually know a peripheral's type using the "peripheral.getType" method.

So when you just enter the command : peripheral.getType("name"),
It does print correctly the name and the type of the peripheral.

However, when you try to plug that result into a table, it only registers the name of the peripheral, not it's actual type. I was expecting it to be registered as a table where index 1 = name and index 2 = type but it doesn't seem to work that way, despite what the CC:T [documentation](https://tweaked.cc/module/peripheral.html#v:getType) claims.

```lua

var = peripheral.getType("minecraft:chest_0")

print(type(var))
      returns : "string"

print(var)
      returns : "minecraft:chest"
```

![Capture d’écran du 2024-12-04 15-42-33](https://github.com/user-attachments/assets/5827733c-c3f6-4cbc-bac7-01f3fc7723a0)

As you can see, it gets registered as a string, not a table. What I want to know is if that peripheral is an "inventory" or not. Am I missing something obvious ?

Thanks in advance for the help,
Martin

**Comment by SquidDev** [comment, marked as answer]
To clarify the documentation, it's the *argument* that can be a table, not the return value. This means you can do both `peripheral.getType("top")` or `peripheral.getType(peripheral.wrap("top"))`.

`getType` will return multiple values — that can be put into a table with `{peripheral.getType("top")}`. You can then loop over that table and look for the `"inventory"` string, but it's probably easier to call `peripheral.hasType("top", "inventory")`.

**Reply by MintTee** [reply]
Hello @SquidDev 

So it was me missing something obvious :sweat_smile: ...

Now it makes total sense ! Indeed, it seems much more straight forward to just use the peripheral.hasType() method.
Thank you very much for the fast answer.
